lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js

```diff
@@ -0,0 +1,214 @@
+      this.apiGatewayRestApiId = restApiId;
+    });
+}
+
+function resolveStage() {
+  const restApiId = this.apiGatewayRestApiId;
+
+  return this.provider.request('APIGateway', 'getStage', {
+    restApiId,
+    stageName: this.options.stage,
+  }).then((res) => {
+    this.apiGatewayStageState = res;
+  }).catch(() => {
+    // fail silently...
+  });
+}
+
+function resolveDeploymentId() {
+  if (_.isEmpty(this.apiGatewayStageState)) {
+    const restApiId = this.apiGatewayRestApiId;
+
+    return this.provider.request('APIGateway', 'getDeployments', {
+      restApiId,
+      limit: 500,
+    }).then(res => {
+      if (res.items.length) {
+        // there will ever only be 1 deployment associated
+        const deployment = res.items.shift();
+        return deployment.id;
+      }
+      return null;
+    })
+    .then((deploymentId) => {
+      this.apiGatewayDeploymentId = deploymentId;
+    });
+  }
+
+  return BbPromise.resolve();
+}
+
+function ensureStage() {
+  if (_.isEmpty(this.apiGatewayStageState)) {
+    const restApiId = this.apiGatewayRestApiId;
+    const deploymentId = this.apiGatewayDeploymentId;
+
+    return this.provider.request('APIGateway', 'createStage', {
+      deploymentId,
+      restApiId,
+      stageName: this.options.stage,
+    });
+  }
+
+  return BbPromise.resolve();
+}
+
+function handleTracing() {
+  const tracing = this.serverless.service.provider.tracing;
+  const tracingEnabled = tracing && tracing.apiGateway;
+
+  let operation = { op: 'replace', path: '/tracingEnabled', value: 'false' };
+  if (tracingEnabled) {
+    operation = { op: 'replace', path: '/tracingEnabled', value: 'true' };
+  }
+  this.apiGatewayStagePatchOperations.push(operation);
+}
+
+function handleLogs() {
+  const provider = this.serverless.service.provider;
+  const logs = provider.logs && provider.logs.restApi;
+  const ops = this.apiGatewayStagePatchOperations;
+
+  let dataTrace = { op: 'replace', path: '/*/*/logging/dataTrace', value: 'false' };
+  let logLevel = { op: 'replace', path: '/*/*/logging/loglevel', value: 'OFF' };
+
+  let operations = [dataTrace, logLevel];
+
+  if (logs) {
+    const service = this.serverless.service.service;
+    const stage = this.options.stage;
+    const region = this.options.region;
+    const logGroupName = `/aws/api-gateway/${service}-${stage}`;
+
+    const destinationArn = {
+      op: 'replace',
+      path: '/accessLogSettings/destinationArn',
+      value: `arn:aws:logs:${region}:${this.accountId}:log-group:${logGroupName}`,
+    };
+    // eslint-disable-next-line max-len
+    const format = { op: 'replace', path: '/accessLogSettings/format', value: 'requestId: $context.requestId, ip: $context.identity.sourceIp, caller: $context.identity.caller, user: $context.identity.user, requestTime: $context.requestTime, httpMethod: $context.httpMethod, resourcePath: $context.resourcePath, status: $context.status, protocol: $context.protocol, responseLength: $context.responseLength' };
```

Comment:
Technically we could just concatenate string, to avoid eslint disabling, or even refer to array join, it'll give us also nice readability of what actually we put into `value`, e.g.:

```javascript
{
  op: 'replace',
  path: '/accessLogSettings/format',
  value: [
    'requestId: $context.requestId',
    'ip: $context.identity.sourceIp',
    'caller: $context.identity.caller',
    'user: $context.identity.user',
    'requestTime: $context.requestTime',
    'httpMethod: $context.httpMethod',
    'resourcePath: $context.resourcePath',
    'status: $context.status',
    'protocol: $context.protocol',
    'responseLength: $context.responseLength'
  ].join(', ')
}
```